apps/vaults/contexts/useSolver.tsx

```diff
@@ -99,23 +101,27 @@ function	WithSolverContextApp({children}: {children: React.ReactElement}): React
 				**************************************************************/
 				if (currentSolver === Solver.WIDO && !isSolverDisabled[Solver.WIDO]) {
 					if (widoQuote.status === 'fulfilled' && widoQuote?.value.raw?.gt(0)) {
+						const requestHash = await hash(JSON.stringify({...request, solver: Solver.WIDO, expectedOut: widoQuote.value.raw.toString()}));
 						performBatchedUpdates((): void => {
-							set_currentSolverState({...wido, quote: widoQuote.value});
+							set_currentSolverState({...wido, quote: widoQuote.value, hash: requestHash});
 							set_isLoading(false);
 						});
```

Comment:
Just a thought, but this also feels very repetitive, perhaps later we can try and extract this somehow.